api/src/entity/friendship.entity.ts

```diff
@@ -0,0 +1,44 @@
+import { UUID } from "crypto";
+import {
+    Entity,
+    Column,
+    PrimaryGeneratedColumn,
+    CreateDateColumn,
+    UpdateDateColumn,
+    ManyToOne,
+    JoinColumn,
+    PrimaryColumn,
+} from 'typeorm';
+import { User } from './user.entity';
+
+export enum FriendshipStatus {
+    PENDING = 'pending',
+    ACCEPTED = 'accepted',
+    DECLINED = 'declined',
+    BLOCKED = 'blocked'
+}
+
+@Entity()
+export class Friendship {
+    @PrimaryColumn()
+    user_id1: number;
+
+    @PrimaryColumn()
+    user_id2: number;
+
+    @Column({
+        type: "enum",
+        enum: FriendshipStatus,
+        default: FriendshipStatus.PENDING
+    })
+    status: FriendshipStatus;
+
+    @Column()
```

Comment:
I tried to use the relational notation:
```
    @ManyToOne(() => User, (user) => user.Id)
    @JoinColumn({ name: "user_id2" })
    user2: User;
```

But it didn't work as I was expecting, I'll be updating this in the future to be use the orm better.